src/devices/bus/gameboy/mbc.cpp

```diff
@@ -4,8 +4,10 @@
 
  Game Boy carts with MBC (Memory Bank Controller)
 
-
- TODO: RTC runs too fast while in-game, in MBC-3 games... find the problem!
+TODO:
+- The MBC3 RTC is not really a true RTC. The chip is used to track passed time even
+  when the console is off, not to keep track of the time of day. Keeping track of
+  passed time when the emulation was not running is currently not supported.
```

Comment:
That’s exactly what an RTC does – it keeps track of time passing when the system is off.  This is exactly what `device_rtc_interface` is for.

Reply:
Nope, it insists to set day-of-month, and this thing has a 0-255 day counter instead. So you'd have to redo set_time at device_start.